Tests/UnitTests/Networking/HTTPClientTests.swift

```diff
@@ -1507,10 +1541,16 @@ private extension BaseHTTPClientTests {
 
     static func extractRequestNumber(from urlRequest: URLRequest) -> Int? {
         do {
-            let requestData = urlRequest.ohhttpStubs_httpBody!
-            let requestBodyDict = try XCTUnwrap(try JSONSerialization.jsonObject(with: requestData,
-                                                                                 options: []) as? [String: Any])
-            return try XCTUnwrap(requestBodyDict[Self.requestNumberKeyName] as? Int)
+            let requestData = try XCTUnwrap(urlRequest.ohhttpStubs_httpBody)
+            let body = try JSONDecoder.default.decode(
+                AnyEncodableRequestBody.self,
+                from: requestData
+            ).body
```

Comment:
I had to refactor these using #2769 because of the change to use `HTTPRequestBody`